### PR TITLE
fix: support double quote string enum

### LINKED ISSUE #97: Doesn't Handle String Union Types with Double Quotes
Example from docs: https://github.com/electron/electron/blob/ce4ae584e357000b324228693ed7556922b6335f/docs/api/app.md?plain=1#L1137-L1141

### LINKED ISSUE #98: Doesn't Handle String Union Types with Quotes inside Backticks
This is definitely a corner case, but currently prevents this example in the docs from being typed properly: https://github.com/electron/electron/blob/ce4ae584e357000b324228693ed7556922b6335f/docs/api/client-request.md?plain=1#L54-L58

I tried single quotes rather than double quotes with the same result, so the issue is quotes inside of the backticks. This might be tricky to make parse correctly with the current regex approach.

### PR DESCRIPTION
**Pull request by MarshallOfSound**
Fixes #97
Fixes #98

The comment in #98 was correct, the regex method here isn't great and is one of the few regex smells we have left in this module. This rewrites our string enum extractor as a token-parser state-machine thing. Hope the comments I left are helpful, I'll note that this method _already_ had really solid tests that helped me catch edge cases and this PR adds even more.

### CODE REVIEW

**Review by ckerr — APPROVED** [review]
+1 on adding the `"`

Also +1 on removing `|`  here; doesn't look like we have any docs that use that notation

**Comment by MarshallOfSound** [comment]
The diff compared to current Electron main:

```diff
diff --git a/electron.d.ts.old b/electron.d.ts
index 4410ccb4bc..f79ddb47af 100644
--- a/electron.d.ts.old
+++ b/electron.d.ts
@@ -2883,7 +2883,7 @@ declare namespace Electron {
      * setting this, the window is still a normal window, not a toolbox window which
      * can not be focused on.
      */
-    setAlwaysOnTop(flag: boolean, level?: 'normal' | 'floating' | 'torn-off-menu' | 'modal-panel' | 'main-menu' | 'status' | 'pop-up-menu' | 'screen-saver', relativeLevel?: number): void;
+    setAlwaysOnTop(flag: boolean, level?: 'normal' | 'floating' | 'torn-off-menu' | 'modal-panel' | 'main-menu' | 'status' | 'pop-up-menu' | 'screen-saver' | 'dock', relativeLevel?: number): void;
     /**
      * Sets the properties for the window's taskbar button.
      *
@@ -5492,7 +5492,7 @@ declare namespace Electron {
      * setting this, the window is still a normal window, not a toolbox window which
      * can not be focused on.
      */
-    setAlwaysOnTop(flag: boolean, level?: 'normal' | 'floating' | 'torn-off-menu' | 'modal-panel' | 'main-menu' | 'status' | 'pop-up-menu' | 'screen-saver', relativeLevel?: number): void;
+    setAlwaysOnTop(flag: boolean, level?: 'normal' | 'floating' | 'torn-off-menu' | 'modal-panel' | 'main-menu' | 'status' | 'pop-up-menu' | 'screen-saver' | 'dock', relativeLevel?: number): void;
     /**
      * Sets the properties for the window's taskbar button.
      *
@@ -18513,7 +18513,7 @@ declare namespace Electron {
      * `strict-origin-when-cross-origin`. Defaults to
      * `strict-origin-when-cross-origin`.
      */
-    referrerPolicy?: string;
+    referrerPolicy?: ('""' | 'no-referrer' | 'no-referrer-when-downgrade' | 'origin' | 'origin-when-cross-origin' | 'unsafe-url' | 'same-origin' | 'strict-origin' | 'strict-origin-when-cross-origin');
     /**
      * can be `default`, `no-store`, `reload`, `no-cache`, `force-cache` or
      * `only-if-cached`.

```

Seems correct

**Review by ckerr — APPROVED** [review]
